File: sentimental-credit/credit.py
try:
    card = input("Card Number: ")
except:
    card = input("Card Number: ")
mult = 0
oth = 0
if (len(card) % 2 == 0):
    for number in range(0, len(card) - 1, 2):
        if ((int(card[number]) * 2) > 9):
            mult += (int(card[number]) * 2) - 9
            # print(mult)
        else:
            mult += int(card[number]) * 2
            # print(mult)
    for num in range(1, len(card), 2):
        oth += int(card[num])
else:
    for i in range(1, len(card) - 1, 2):
        if ((int(card[i]) * 2) > 9):
            mult += (int(card[i]) * 2) - 9
            # print(mult)
        else:
            mult += int(card[i]) * 2
            # print(mult)
    for b in range(0, len(card), 2):
        oth += int(card[b])
# print(oth)
# print(mult)
sum1 = oth + mult
# print(sum1)

if (sum1 % 10 == 0):
    if (card[0] == "3" and (card[1] == "7" or card[1] == "4")):
        print("AMEX")
    elif (card[0] == "5" and (card[1] == "1" or card[1] == "2" or card[1] == "3" or card[1] == "4" or card[1] == "5")):
        print("MASTERCARD")
    elif (card[0] == "4"):
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int num1 = 0;
    int num2 = 0;
    long number = get_long("Enter credit card number: ");
    int count = 1;
    int num3 = 0;
    int card = 0;
    long ten = 10;
    while ((card == 0) && ((number / ten) > 0))
    {
        if ((number / ten == 34) || (number / ten == 37))
        {
            card = 1;
        }
        else if ((number / ten == 51) || (number / ten == 52) || (number / ten == 53) || (number / ten == 54) || (number / ten == 55))
        {
            card = 2;
        }
        else if (number / ten == 4)
        {
            card = 3;
        }
        else
        {
            ten = ten * 10;
        }
    }
    while (number > 0)
    {
        if (count == 1)
        {
            num1 += number % 10;
            number = number / 10;
            count -= 1;
        }
        else
        {
            if (((number % 10) * 2) > 9)
            {
                int num5 = ((number % 10) * 2) - 9;
                num2 = num5 + num2;
                number = number / 10;
                count += 1;
            }
            else
            {
                num2 += (number % 10) * 2;
                number = number / 10;
                //num2 += (number % 10) * 2;
                //number = number / 10;
                count += 1;
            }
        }
    }
    num3 = num1 + num2;
    if ((num3 % 10 == 0) && (card == 1) && (ten > 100000000000))
    {
        printf("AMEX\n");
    }
    else if ((num3 % 10 == 0) && (card == 2) && (ten > 100000000000))
    {
        printf("MASTERCARD\n");
    }
    else if ((num3 % 10 == 0) && (card == 3) && (ten > 100000000000))
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }

}